low-level/framework/utils/iem.py

```diff
@@ -155,6 +153,12 @@ def check_exsisting_fault_iems(self):
             if prev_fault_iem_event:
                 self.fault_iems.append(event_name)
 
-def generate_iem(iem):
-    """Generate iem."""
-    syslog.syslog(iem)
+    @classmethod
+    def generate_iem(cls, module, event_code, severity, description):
```

Comment:
why do we need a classmethod? we can keep this as a staticmethod?

Reply:
Ack